loopy/transform/callable.py

```diff
@@ -632,6 +632,9 @@ def map_call(self, expr):
     def map_constant(self, expr):
         return False
 
+    def map_type_cast(self, expr):
+        return False
```

Comment:
This needs to recurse into the children of the type cast, IIUC. (There might be a function call in there after all.)

Reply:
Committed to pyop2/codegen/loopycompat.py in OP2/PyOP2#612  as https://github.com/OP2/PyOP2/pull/612/commits/bf627db8e2f3381e42d39b74ba013ea84135583f